js/src/ipc/writer/binary.ts

```diff
@@ -66,15 +66,15 @@ export function* serializeFile(table: Table) {
     for (const [id, field] of table.schema.dictionaries) {
         const vec = table.getColumn(field.name) as DictionaryVector;
         if (vec && vec.dictionary) {
-            ({ metadataLength, buffer } = serializeDictionaryBatch(vec.dictionary, id));
-            dictionaryBatches.push(new FileBlock(metadataLength, buffer.byteLength, byteLength));
+            ({ metadataLength, bodyLength, buffer } = serializeDictionaryBatch(vec.dictionary, id));
+            dictionaryBatches.push(new FileBlock(metadataLength, bodyLength, byteLength));
             byteLength += buffer.byteLength;
             yield buffer;
         }
     }
     for (const recordBatch of table.batches) {
-        ({ metadataLength, buffer } = serializeRecordBatch(recordBatch));
-        recordBatches.push(new FileBlock(metadataLength, buffer.byteLength, byteLength));
+        ({ metadataLength, bodyLength, buffer } = serializeRecordBatch(recordBatch));
+        recordBatches.push(new FileBlock(metadataLength, bodyLength, byteLength));
```

Comment:
Could you explain this a little bit?
It seems unrelated to the associated JIRA since it's a change in the writer and not the reader - is this a fix for another bug?

Reply:
Yeah, it's for https://issues.apache.org/jira/browse/ARROW-3256

Reply:
Ah gotcha. I'll just add that jira to the description and otherwise this looks good